src/main/java/org/opentripplanner/routing/impl/HighestFareInFreeTransferWindowFareServiceImpl.java

```diff
@@ -0,0 +1,82 @@
+import org.slf4j.LoggerFactory;
+
+public class HighestFareInFreeTransferWindowFareServiceImpl extends DefaultFareServiceImpl {
+    private final boolean analyzeInterlinedTransfers;
+    private final int freeTransferWindowInMinutes;
+
+    public HighestFareInFreeTransferWindowFareServiceImpl(
+        Collection<FareRuleSet> regularFareRules, int freeTransferWindowInMinutes, boolean analyzeInterlinedTransfers
+    ) {
+        addFareRules(FareType.regular, regularFareRules);
+        this.freeTransferWindowInMinutes = freeTransferWindowInMinutes;
+        this.analyzeInterlinedTransfers = analyzeInterlinedTransfers;
+    }
+
+    private static final long serialVersionUID = 20120229L;
+    @SuppressWarnings("unused")
+    private static final Logger LOG = LoggerFactory.getLogger(HighestFareInFreeTransferWindowFareServiceImpl.class);
+
+    /**
+     * The fare calculation is designed to charge the rider incrementally as they use each service. The total cost of
+     * the itinerary will be equal to the leg of the journey that had the maximum fare. Additionally, a free transfer
+     * window is taken into account that will apply the current highest cost within the free transfer window and then
+     * reset to a new window and calculate additional free transfers from there.
+     */
+    @Override
+    protected boolean populateFare(Fare fare, Currency currency, FareType fareType, List<Ride> rides,
+                                   Collection<FareRuleSet> fareRules) {
+        float cost = 0;
+        float currentTransferWindowCost = 0;
+        long freeTransferWindowEndTime = -1;
+        for (Ride ride : rides) {
+            List<Ride> leg = new ArrayList<>();
+            leg.add(ride);
+            float rideCost = calculateCost(fareType, leg, fareRules);
+
+            if (ride.startTime > freeTransferWindowEndTime) {
+                // free transfer window has expired, add to overall cost
+                cost += currentTransferWindowCost;
+                // reset current window cost
+                currentTransferWindowCost = 0;
+                // reset transfer window end time to trigger recalculation in next block
+                freeTransferWindowEndTime = -1;
+            }
+
+            // recalculate free transfer window if needed
+            if (freeTransferWindowEndTime == -1) {
+                freeTransferWindowEndTime = ride.startTime + freeTransferWindowInMinutes * 60;
+            }
+
+            currentTransferWindowCost = Float.max(currentTransferWindowCost, rideCost);
+        }
+        cost += currentTransferWindowCost;
+        if (cost < Float.POSITIVE_INFINITY) fare.addFare(fareType, getMoney(currency, cost));
+        return cost > 0 && cost < Float.POSITIVE_INFINITY;
+    }
+
+    /**
+     * Returns true if configured to analyze interlined transfers and the previous edge was an interline transfer.
+     */
+    @Override
+    protected boolean createRideDueToInterlining(State state, Ride ride) {
```

Comment:
Would a better name for the method be `treatInterlinedTripsAsSeparateRides`? Perhaps not, but maybe that's a good name for the `analyzeInterlinedTransfers` field.

Reply:
It's such a complex thing that I don't think any naming is going to effectively communicate it.